gc/gc-base/src/main/java/org/projectnessie/gc/base/GCImpl.java

```diff
@@ -0,0 +1,186 @@
+      DistributedIdentifyContents distributedIdentifyContents =
+          new DistributedIdentifyContents(session, gcParams);
+      List<Reference> liveReferences = api.getAllReferences().get().getReferences();
+      Map<Reference, Instant> droppedReferenceTimeMap = collectDeadReferences(api);
+      List<Reference> allRefs = new ArrayList<>(liveReferences);
+      if (droppedReferenceTimeMap.size() > 0) {
+        allRefs.addAll(droppedReferenceTimeMap.keySet());
+      }
+      long totalCommitsInDefaultReference = getTotalCommitsInDefaultReference(api);
+      // Identify the live contents and return the bloom filter per content-id
+      Map<String, ContentBloomFilter> liveContentsBloomFilterMap =
+          distributedIdentifyContents.getLiveContentsBloomFilters(
+              allRefs, totalCommitsInDefaultReference, droppedReferenceTimeMap);
+      // Identify the expired contents
+      return distributedIdentifyContents.getIdentifiedResults(liveContentsBloomFilterMap, allRefs);
+    }
+  }
+
+  private long getTotalCommitsInDefaultReference(NessieApiV1 api) {
+    ReferenceMetadata defaultRefMetadata = null;
+    try {
+      defaultRefMetadata =
+          api.getReference()
+              .refName(api.getDefaultBranch().getName())
+              .fetch(FetchOption.ALL)
+              .get()
+              .getMetadata();
+    } catch (NessieNotFoundException ex) {
+      throw new RuntimeException(ex);
+    }
+    return (defaultRefMetadata != null && defaultRefMetadata.getNumTotalCommits() != null)
+        ? defaultRefMetadata.getNumTotalCommits()
+        : 0;
+  }
+
+  private static Map<Reference, Instant> collectDeadReferences(NessieApiV1 api) {
+    Map<Reference, Instant> droppedReferenceTimeMap = new HashMap<>();
+    Stream<RefLogResponse.RefLogResponseEntry> reflogStream;
+    try {
+      reflogStream =
+          StreamingUtil.getReflogStream(
+              api,
+              null,
+              null,
+              String.format(
+                  "reflog.operation == '%s' || reflog.operation == " + "'%s'",
+                  RefLogResponse.RefLogResponseEntry.DELETE_REFERENCE,
+                  RefLogResponse.RefLogResponseEntry.ASSIGN_REFERENCE),
+              OptionalInt.empty());
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+    reflogStream.forEach(
+        entry -> {
+          String hash;
+          switch (entry.getOperation()) {
+            case RefLogResponse.RefLogResponseEntry.DELETE_REFERENCE:
+              hash = entry.getCommitHash();
+              break;
+            case RefLogResponse.RefLogResponseEntry.ASSIGN_REFERENCE:
+              hash = entry.getSourceHashes().get(0);
+              break;
+            default:
+              throw new RuntimeException(
+                  entry.getOperation() + " operation found in dead reflog query");
+          }
+          switch (entry.getRefType()) {
```

Comment:
I'd prefer to do the switch in `RefUtil` and only do `droppedReferenceTimeMap.put(RefUtil.fromRefLogEntry(entry), getInstantFromMicros(entry.getOperationTime()))` here.

I'm sure that I've mentioned `RefUtil` before

Reply:
I have replied to that saying it is a new module dependency just for one if check. Is it really need to do ?
but I didn't get reply

Reply:
Ah, right - RefUtil isn't available here.